### PR TITLE
refactor: move data generator to IOx repo and fix build

### PR DESCRIPTION
**Pull request by pauldix**
Just a lift and shift from the other repo along with a bunch of little changes to fix imports and builld.

### CODE REVIEW

**Review by alamb — APPROVED** [review]
I am not sure exactly how much review is needed on this PR as it seems to be one large commit  (and I can't say I reviewed the entire commit Screen Shot 2021-08-19 at 1 32 55 PM 👀  👀  . I know this is largely just a copy, but those are some great stats 🦾 😆 )

Anyhow, the only other thought I had if that  we should **archive** (turn read only) and add a note to the old https://github.com/influxdata/iox_data_generator repo so people don't get confused about which repo is active

File: iox_data_generator/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+[package]
+name = "data_generator"
```
Comment (carols10cents) [inline iox_data_generator/Cargo.toml:2]: It's somewhat expected that the crate directory and package name match, so I would make this:

```suggestion
name = "iox_data_generator"
```

File: iox_data_generator/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+version = "0.1.0"
+authors = ["Paul Dix <paul@pauldix.net>"]
+edition = "2018"
+default-run = "data_generator"
```
Comment (carols10cents) [inline iox_data_generator/Cargo.toml:6]: And this: 

```suggestion
default-run = "iox_data_generator"
```

File: iox_data_generator/Cargo.lock [hunk 3]
```diff
@@ -0,0 +1,2862 @@
+# This file is automatically @generated by Cargo.
```
Comment (carols10cents) [inline iox_data_generator/Cargo.lock:1]: This actually shouldn't be moved over; now that it's in the workspace, it'll use the top-level Cargo.lock.

File: iox_data_generator/README.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+```
+cargo build --release
+```
+
+And the built binary has command line help:
+
+```
+./target/release/data_generator --help
+```
```
Comment (carols10cents) [inline iox_data_generator/README.md:16]:
```suggestion
# `iox_data_generator`

The `iox_data_generator` tool creates random data points according to a specification and loads them
into an `iox` instance to simulate real data.

To build and run, [first install Rust](https://www.rust-lang.org/tools/install). Then from the root of the `influxdb_iox` repo run:

```
cargo build -p iox_data_generator --release
```

And the built binary has command line help:

```
./target/release/iox_data_generator --help
```
```

File: iox_data_generator/README.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+## Use with two IOx servers and Kafka
+
+The data generator tool can be used to simulate data being written to IOx in various shapes. This
+is how to set up a local experiment for profiling or debugging purposes using a database in two IOx
+instances: one writing to Kafka and one reading from Kafka.
+
+If you're profiling IOx, be sure you've compiled and are running a release build using either:
+
+```
+# in your influxdb_iox checkout
```
Comment (carols10cents) [inline iox_data_generator/README.md:29]: These comments about which checkout to be in aren't needed anymore

File: iox_data_generator/README.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+Once you have the two IOx servers and one Kafka instance running, create a database with a name in
+the format `[orgname]_[bucketname]`. For example, create a database in IOx named `mlb_pirates`, and
+the org you'll use in the data generator will be `mlb` and the bucket will be `pirates`. The
+`DatabaseRules` defined in `src/bin/create_database.rs` will set up a database in the "writer" IOx
+instance to write to Kafka and the database in the "reader" IOx instance to read from Kafka if
+you run it with:
+
+```
+# in your iox_data_generator checkout
+cargo run --bin create_database -- --writer 127.0.0.1:8082 --reader 127.0.0.1:8086 mlb_pirates
```
Comment (carols10cents) [inline iox_data_generator/README.md:75]:
```suggestion
cargo run -p iox_data_generator --bin create_database -- --writer 127.0.0.1:8082 --reader 127.0.0.1:8086 mlb_pirates
```

File: iox_data_generator/README.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+Once the database is created, decide what kind of data you would like to send it. You can use an
+existing data generation schema in the `schemas` directory or create a new one, perhaps starting
+from an existing schema as a guide. In this example, we're going to use `schemas/cap-write.toml`.
+
+Next, run the data generation tool as follows:
+
+```
+# in your iox_data_generator checkout
+cargo run -- --spec schemas/cap-write.toml --continue --host 127.0.0.1:8080 --token arbitrary --org mlb --bucket pirates
```
Comment (carols10cents) [inline iox_data_generator/README.md:89]:
```suggestion
cargo run -p iox_data_generator -- --spec iox_data_generator/schemas/cap-write.toml --continue --host 127.0.0.1:8080 --token arbitrary --org mlb --bucket pirates
```

File: iox_data_generator/README.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+```
+
+- `--spec schemas/cap-write.toml` sets the schema you want to use to generate the data
```
Comment (carols10cents) [inline iox_data_generator/README.md:92]:
```suggestion
- `--spec iox_data_generator/schemas/cap-write.toml` sets the schema you want to use to generate the data
```

File: iox_data_generator/benches/point_generation.rs [hunk 9]
```diff
@@ -0,0 +1,66 @@
+use criterion::{criterion_group, criterion_main, Criterion, Throughput};
+use data_generator::{
```
Comment (carols10cents) [inline iox_data_generator/benches/point_generation.rs:2]: When the package name is changed, this'll need to be changed too:

```suggestion
use iox_data_generator::{
```

File: iox_data_generator/src/main.rs [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+    missing_copy_implementations,
+    missing_debug_implementations,
+    clippy::explicit_iter_loop,
+    clippy::use_self
+)]
+
+use chrono::prelude::*;
+use chrono_english::{parse_date_string, Dialect};
+use clap::{crate_authors, crate_version, App, Arg};
+use data_generator::{specification::DataSpec, write::PointsWriterBuilder};
```
Comment (carols10cents) [inline iox_data_generator/src/main.rs:12]:
```suggestion
use iox_data_generator::{specification::DataSpec, write::PointsWriterBuilder};
```

File: iox_data_generator/src/main.rs [hunk 11]
```diff
@@ -0,0 +1,265 @@
...
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    tracing_subscriber::fmt::init();
+
+    let help = r#"IOx data point generator
+
+Examples:
+    # Generate data points using the specification in `spec.toml` and save in the `lp` directory
+    data_generator -s spec.toml -o lp
```
Comment (carols10cents) [inline iox_data_generator/src/main.rs:23]: These all through here should be changed to `iox_data_generator`

File: iox_data_generator/src/main.rs [hunk 12]
```diff
@@ -0,0 +1,265 @@
...
+            matches.is_present("create"),
+            matches.value_of("ORG_ID"),
+        );
+
+        PointsWriterBuilder::new_api(host, org, bucket, token, create_bucket, org_id).await?
+    } else {
+        panic!("One of --output or --host must be provided.");
+    };
+
+    let result = data_generator::generate::<rand::rngs::SmallRng>(
```
Comment (carols10cents) [inline iox_data_generator/src/main.rs:173]:
```suggestion
    let result = iox_data_generator::generate::<rand::rngs::SmallRng>(
```

**Review by carols10cents — CHANGES_REQUESTED** [review]
A few tweaks to the names to be consistent with the new location, and the Cargo.lock file doesn't need to come over :)

**Comment by alamb** [comment]
Updating / rebasing against `main` will fix the tests failure (get https://github.com/influxdata/influxdb_iox/pull/2351)

I think clippy is being a bit more persickety now than in times past

**Comment by pauldix** [comment]
@carols10cents ok, fixed all in amended commit, rebased and force pushed. I think it's good to go now.

**Review by carols10cents — APPROVED** [review]

**Comment by alamb** [comment]
🎉

**Pull request merged**